source/usage-examples/updateOne.txt

```diff
@@ -0,0 +1,118 @@
+=================
+Update a Document
+=================
+
+.. default-domain:: mongodb
+
+You can update a single document using the ``updateOne()`` method on
+a ``MongoCollection`` object. The method accepts a **filter** that matches the
+document you want to update and an **update** statement that instructs the
+driver to change the matching document. The ``updateOne()`` method only
+updates the first document that matches the filter.
+
+To perform an update with the ``updateOne()`` method, you must pass
+a query filter and an update document. The query filter specifies the criteria
+for which document to perform the update on and the update document provides
+instructions on what changes to make to it.
+
+You can optionally pass an instance of ``UpdateOptions`` to the method in
+order to specify the behavior. For example, if you set the ``upsert`` field of
+the ``UpdateOptions`` object to ``true``, the operation inserts a new
+document from the update document if no documents match the query filter.
+See the link to the ``UpdateOptions`` API documentation at the bottom of
+this page for more information.
+
+The ``updateOne()`` method accepts additional options for the call in an
+``UpdateOptions`` object. You can use the ``upsert`` field in this object to
+specify that MongoDB should create a new document from the update fields if
+no documents match the query filter.
+
+Upon successful execution, the ``updateOne()`` method returns an instance
+of ``UpdateResult``.  You can retrieve information such as the number of
+documents modified calling the ``getModifiedCount()`` method, or the
+value of ``_id`` field by calling the ``getUpsertedId()`` method if you
+specified ``upsert(true)`` in the ``UpdateOptions``.
+
+If your update operation fails, the driver raises an exception.
+For example, if you try to set a value for the immutable field ``_id`` in
+your update document, the method throws a ``MongoWriteException`` with the
+message:
+
+.. code-block:: none
+
+   Performing an update on the path '_id' would modify the immutable field '_id'
+
+If your update document contains a change that violates unique index
+rules, the method throws a ``DuplicateKeyException`` with the error
```

Comment:
The CRUD API does not throw DuplicateKeyException.  It throws a MongoWriteException.  This is in the Javadoc.